### DISCUSSION TITLE
Form submit with input file - Next js + craft cms headless

### DISCUSSION THREAD

**Discussion by kaduppg** [craftcms/cms#10133] [Q&A] [answered]
Hey guys, I was wondering what is the best solution when it comes to submit a form with a file input? Should I use mutation or create a controller? If so ... how can i link the file uploaded with the assets ? If there's any tutorial please let me know.

Cheers.

**Comment by brandonkelly** [comment, marked as answer]
Here’s our docs for it: https://craftcms.com/docs/3.x/graphql.html#saving-files-via-mutations
